src/middleware/middleware.builders.js

```diff
@@ -114,6 +121,53 @@ export async function fetchManyFn (req, res, next) {
   }
 }
 
+export async function fetchOneFromAllDatasetsFn (req, res, next) {
+  try {
+    const query = this.query({ req, params: req.params })
+    const promises = availableDatasets.map((dataset) => {
+      return datasette.runQuery(query, dataset)
+    })
+    const result = await Promise.all(promises)
+    req[this.result] = Object.fromEntries(
+      result.reduce((acc, { formattedData }, i) => {
+        if (formattedData.length > 0) {
+          acc.push([availableDatasets[i], formattedData[0]])
+        }
+        return acc
+      }, [])
+    )
+    logger.debug({ type: types.DataFetch, message: 'fetchOneFromAllDatasets', resultKey: this.result })
+    next()
+  } catch (error) {
+    logger.debug('fetchMany: failed', { type: types.DataFetch, errorMessage: error.message, endpoint: req.originalUrl, resultKey: this.result })
+    req.handlerName = `fetching '${this.result}'`
+    next(error)
+  }
+}
+
+export async function fetchManyFromAllDatasetsFn (req, res, next) {
+  try {
+    const query = this.query({ req, params: req.params })
+    const promises = availableDatasets.map((dataset) => {
+      return datasette.runQuery(query, dataset).catch(error => {
+        logger.error('Query failed for dataset', { dataset, error, type: types.DataFetch })
+        return { formattedData: [] }
```

Comment:
again, logging the exception object itself works differently than printing it in the console etc. You need to manually extract the data you want to see in the logs.